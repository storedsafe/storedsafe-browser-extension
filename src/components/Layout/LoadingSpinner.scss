@import '../../variables';

.loading-spinner {
  position: relative;
  height: 3em;
  width: 3em;
  margin: auto;
}

.spinner {
  &::after,
  &::before {
    border: 0.2em solid transparent;
    border-radius: 100%;
    border-top-color: $color-accent-0;
    box-sizing: border-box;
    content: '';
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 100%;
  }

  &::before {
    animation: spin 1s infinite;
    transition: all $transition-duration*3;
    z-index: 100;
  }

  &::after {
    background-image: url('../../../graphics/padlock.svg');
    background-size: contain;
    border-color: $color-primary-1;
    transition: all $transition-duration*3;
  }

  &.error::before,
  &.warning::before,
  &.success::before {
    animation: none;
    border: 0;
  }

  &.error::after {
    background-image: url('../../../graphics/padlock-error.svg');
    border-color: $color-error-0;
  }

  &.warning::after {
    background-image: url('../../../graphics/padlock-warning.svg');
    border-color: $color-warning-0;
  }

  &.success::after {
    background-image: url('../../../graphics/padlock-unlocked.svg');
    border-color: $color-accent-0;
  }
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
