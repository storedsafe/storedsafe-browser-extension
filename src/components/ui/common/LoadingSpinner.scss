@import '../../../variables';

.loading-spinner {
  $size: 3em;

  position: relative;
  height: $size;
  width: $size;
  margin: auto;

  .spinner {
    &::after,
    &::before {
      position: absolute;
      content: '';
      border-radius: 100%;
      box-sizing: border-box;
      left: 0;
      top: 0;
      height: 100%;
      width: 100%;
    }

    &::after {
      border: 0.3em solid #fff;
      border-top-color: $color-accent-0;
      animation: spin 1s infinite;
      transition: all $transition-duration*3;
      z-index: 100;
    }

    &::before {
      background-image: url('../../../graphics/padlock.svg');
      background-size: contain;
      border: 0;
      transition: all $transition-duration*3;
    }

    &.error::after,
    &.warning::after,
    &.success::after {
      animation: none;
      border: 0;
    }

    &.error::before {
      background-image: url('../../../graphics/padlock-error.svg');
      border-color: $color-danger-0;
    }

    &.warning::before {
      background-image: url('../../../graphics/padlock-warning.svg');
      border-color: $color-warning-0;
    }

    &.success::before {
      background-image: url('../../../graphics/padlock-unlocked.svg');
      border-color: $color-accent-0;
    }
  }
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
