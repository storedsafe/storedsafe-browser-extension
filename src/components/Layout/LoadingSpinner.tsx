import React from 'react';
import './LoadingSpinner.scss';

interface LoadingSpinnerProps {
  status?: 'loading' | 'success' | 'error' | 'warning';
}

export const LoadingSpinner: React.SFC<LoadingSpinnerProps> = ({
  status,
}: LoadingSpinnerProps) => (
  <div className="loading-spinner">
    <div className={`spinner ${status}`} />
  </div>
);

LoadingSpinner.defaultProps = {
  status: 'loading',
};
