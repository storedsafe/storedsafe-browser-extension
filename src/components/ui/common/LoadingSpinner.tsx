import React from 'react';
import './LoadingSpinner.scss';

interface LoadingSpinnerProps {
  status?: 'loading' | 'success' | 'error' | 'warning';
}

export const LoadingSpinner: React.FunctionComponent<LoadingSpinnerProps> = ({
  status,
}: LoadingSpinnerProps) => (
  <div className="loading-spinner">
    <div className={`spinner ${status}`} />
  </div>
);

LoadingSpinner.defaultProps = {
  status: 'loading',
};
