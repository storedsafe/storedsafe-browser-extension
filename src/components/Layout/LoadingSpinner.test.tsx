import * as React from 'react';
import { shallow } from 'enzyme';
import { LoadingSpinner } from './LoadingSpinner';

test('<LoadingSpinner />', () => {
  const wrapper = shallow(<LoadingSpinner />);
  expect(wrapper).toMatchSnapshot();
});

test('<LoadingSpinner loading />', () => {
  const wrapper = shallow(<LoadingSpinner status="loading" />);
  expect(wrapper).toMatchSnapshot();
});

test('<LoadingSpinner success />', () => {
  const wrapper = shallow(<LoadingSpinner status="success" />);
  expect(wrapper).toMatchSnapshot();
});

test('<LoadingSpinner warning />', () => {
  const wrapper = shallow(<LoadingSpinner status="warning" />);
  expect(wrapper).toMatchSnapshot();
});

test('<LoadingSpinner error />', () => {
  const wrapper = shallow(<LoadingSpinner status="error" />);
  expect(wrapper).toMatchSnapshot();
});
