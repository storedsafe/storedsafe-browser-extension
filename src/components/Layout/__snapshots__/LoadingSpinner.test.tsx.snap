// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> 1`] = `ShallowWrapper {}`;

exports[`<LoadingSpinner error /> 1`] = `ShallowWrapper {}`;

exports[`<LoadingSpinner loading /> 1`] = `ShallowWrapper {}`;

exports[`<LoadingSpinner success /> 1`] = `ShallowWrapper {}`;

exports[`<LoadingSpinner warning /> 1`] = `ShallowWrapper {}`;
